### DISCUSSION TITLE
Mep modeler from Python code

### DISCUSSION THREAD

**Discussion by Isaina** [IfcOpenShell/IfcOpenShell#3456] [Ideas]
Hi you add mep modeler ifcpipe and ifcduct to ifcopenshell to model and summarize Mep installation from python code some thing like ifcpipe start coordinations x,y,z and direction with length . Or write mep modeler for ifc open shell

**Comment by Moult** [comment]
This isn't too different from a previous discussion we had about how to do wall from 2 points (start / end).

However for MEP it's a little difference since whereas with walls you worry about wall joints (butt, mitre, etc) for MEP you have to worry about fittings. There are a lot of ways to do fittings and we don't yet have code for it. However if all you're after is the pipe / duct segment with a start / end that's pretty trivial.

**Reply by Isaina** [reply]
Fittings are defined in ifc https://standards.buildingsmart.org/IFC/RELEASE/IFC2x3/TC1/HTML/ifchvacdomain/lexical/ifcductfittingtype.htm

https://standards.buildingsmart.org/IFC/RELEASE/IFC2x3/TC1/HTML/ifchvacdomain/lexical/ifcductfittingtype.htm

Because when you create pipe or duct this types are defined in ifc but in higher LOD (level of details) you can define producer and geometry but generic would be fine. 

Something like two functions

pipe (start point, finish point, type(like duct, fitting, terminal etc))
Duct (start point, finish point , type(like duct, fitting, terminal etc))

also check that start and finish x,y,z plains only one can be different because pipes are not flexible for eg, x,y are fixed and z must be different
